hedera-node/hedera-app/src/main/java/com/hedera/node/app/state/merkle/MerkleSchemaRegistry.java

```diff
@@ -216,16 +217,27 @@ public void migrate(
             remainingStates.removeAll(statesToRemove);
             final var newStates = new FilteredWritableStates(writeableStates, remainingStates);
 
-            final var migrationContext =
-                    new MigrationContextImpl(previousStates, newStates, config, networkInfo, genesisRecordsBuilder);
+            // For any changes to state that depend on other services outside the current service, we need a reference
+            // to the overall state that we can pass into the context. This reference to overall state will be strictly
+            // controlled via the MigrationContext API so that only changes explicitly specified in the interface can be
+            // made (instead of allowing any arbitrary change to overall state). As above, we won't commit anything
+            // until after this service's migration
+            final var migrationStack = new SavepointStackImpl(hederaState);
+
+            final var migrationContext = new MigrationContextImpl(
+                    previousStates, newStates, config, networkInfo, genesisRecordsBuilder, migrationStack);
             if (updateInsteadOfMigrate) {
                 schema.restart(migrationContext);
             } else {
                 schema.migrate(migrationContext);
             }
+            // Now we commit all the service-specific changes made during this service's update or migration
             if (writeableStates instanceof MerkleHederaState.MerkleWritableStates mws) {
                 mws.commit();
             }
+            // And commit any changes that were made to any state belonging to other services
+            migrationStack.createSavepoint();
+            migrationStack.commit();
```

Comment:
This will have no effect. You do not need to create a save-point and you probably want to use `migrationStack.commitFullStack()`. `commit()` will only commit the top-most frame on the stack.